app/services/api/v1x0/catalog/portfolio_item_orderable.rb

```diff
@@ -0,0 +1,49 @@
+module Api
+  module V1x0
+    module Catalog
+      class PortfolioItemOrderable
+        attr_reader :result
+
+        def initialize(portfolio_item)
+          @result = false
+          @portfolio_item = portfolio_item
+        end
+
+        def process
+          @result = !archived? && !survey_changed? && source_available?
```

Comment:
@mkanoor I think we discussed not only determining that it was broken but also the reason why.  Should we test all the conditions and add an accessor to return results?

Also, I think we have to handle when the source is actually deleted in addition to the "available" check.

Reply:
@gmcculloug I have added a messages method to list all the errors that we encounter. But from a user_capabilities perspective we only have a boolean field to convey to the UI that the product is not order able, there is no field that lets us pass in why its not order able.